src/devices/Nmea0183/Sentences/TransducerDataSet.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using UnitsNet;
+
+namespace Iot.Device.Nmea0183.Sentences
+{
+    /// <summary>
+    /// A single data set from an XDR sentence
+    /// </summary>
+    public sealed class TransducerDataSet
+    {
+        /// <summary>
+        /// Creates an empty instance
+        /// </summary>
+        public TransducerDataSet()
+        {
+            Unit = string.Empty;
+            Value = 0;
+            DataType = string.Empty;
+            DataName = string.Empty;
+        }
+
+        /// <summary>
+        /// Creates an instance filled with data.
+        /// </summary>
+        public TransducerDataSet(string dataType, double value, string unit, string dataName)
+        {
+            DataType = dataType;
+            Value = value;
+            Unit = unit;
+            DataName = dataName;
+        }
+
+        /// <summary>
+        /// The unit of the measurement
+        /// Known values:
+        /// - B (bar)
+        /// - P (Pascal, Percentage)
+        /// - D (degrees)
+        /// - G (Acceleration, g-Force)
+        /// - C (degrees Celsius)
+        /// - M (Mass, Liters)
+        /// - V (Voltage)
+        /// - A (Ampere)
+        /// </summary>
+        public string Unit { get; set; }
```

Comment:
Would it make sense to have an enum rather than a 1 length string with those know values?
Similar question for the rest. It doesn't prevent to work if there is another unknown, you'll still get the value anyway

Reply:
The 1-character string is used in the message protocol itself, and since the values are not even unique (e.g. P can be pascals as well as percentage), I'd rather leave the interpretation to the client.